### PR TITLE
🔥 Add Cache Invalidation Option to Cache Middleware

### PR DESCRIPTION
**Pull request by hcancelik**
## Description

This is a new option for the cache middleware to invalidate the cache. This is based on the discussion at https://github.com/gofiber/fiber/issues/3034

## Changes introduced

- [ ] Benchmarks: Describe any performance benchmarks and improvements related to the changes.
- [x] Documentation Update: Detail the updates made to the documentation and links to the changed files.
- [x] Changelog/What's New: Include a summary of the additions for the upcoming release notes.
- [ ] Migration Guide: If necessary, provide a guide or steps for users to migrate their existing code to accommodate these changes.
- [ ] API Alignment with Express: Explain how the changes align with the Express API.
- [ ] API Longevity: Discuss the steps taken to ensure that the new or updated APIs are consistent and not prone to breaking changes.
- [x] Examples: Provide examples demonstrating the new features or changes in action.

## Type of change

- [x] Enhancement (improvement to existing features and functionality)

## Checklist

- [ ] Followed the inspiration of the Express.js framework for new functionalities, making them similar in usage.
- [x] Conducted a self-review of the code and provided comments for complex or critical parts.
- [x] Updated the documentation in the `/docs/` directory for [Fiber's documentation](https://docs.gofiber.io/).
- [x] Added or updated unit tests to validate the effectiveness of the changes or new features.
- [x] Ensured that new and existing unit tests pass locally with the changes.
- [ ] Verified that any new dependencies are essential and have been agreed upon by the maintainers/community.
- [ ] Aimed for optimal performance with minimal allocations in the new code.
- [ ] Provided benchmarks for the new code to analyze and improve upon.

### CODE REVIEW

**Comment by gaby** [comment]
@hcancelik I think the name doesnt make it clear what it is, it could be a func or param. Something more appropiate would be:
- InvalidateCacheFunc
- CacheInvalidator
- CacheInvalidationFunc

**Comment by hcancelik** [comment]
@gaby appreciate for the feedback. I have renamed it to `CacheInvalidator`.

File: docs/middleware/cache.md [hunk 1]
```diff
@@ -62,22 +62,33 @@ app.Get("/", func(c fiber.Ctx) error {
...
 
+You can also invalidate the cache by using the `CacheInvalidator` function as shown below:
+
+```go
+app.Use(cache.New(cache.Config{
+    CacheInvalidator: func(c fiber.Ctx) bool {
+        return fiber.Query[bool](c, "invalidateCache")
+    },
+}))
+```
```
[the remark this answers is not in the document]
Comment (gaby) [reply docs/middleware/cache.md:73]: @hcancelik can you add the suggestion above

[the remark this answers is not in the document]
Comment (hcancelik) [reply docs/middleware/cache.md:73]: @gaby Sure. Would this be sufficient enough?

File: docs/middleware/cache.md [hunk 2]
```diff
@@ -62,31 +62,44 @@ app.Get("/", func(c fiber.Ctx) error {
...
 ## Default Config
 
 ```go
 var ConfigDefault = Config{
     Next:         nil,
     Expiration:   1 * time.Minute,
-	CacheHeader:  "X-Cache",
+    CacheHeader:  "X-Cache",
     CacheControl: false,
+    CacheInvalidator: nil,
```
[the remark this answers is not in the document]
Comment (hcancelik) [reply docs/middleware/cache.md:103]: There are no hard tabs on line 103.

**Review by gaby — APPROVED** [review]
LGTM

**Pull request merged**